docs/index-server.md

```diff
@@ -0,0 +1,128 @@
+
+## Reallocation of executor
+In case executor(s) become dead/unavailable then the segments that were
+earlier being handled by those would be reassigned to some other
+executor using the distribution logic.
+
+**Note:** Cache loading would be done again in the new executor for the
+current query.
+
+## Fallback
+In case of any failure the index server would fallback to embedded mode
+which means that the JDBCServer would take care of distributed pruning.
+A similar job would be fired by the JDBCServer which would take care of
+pruning using its own executors. If for any reason the embedded mode
+also fails to prune the datamaps then the job would be passed on to
+driver.
+
+
+## Configurations
+
+##### carbon.properties 
+
+| Name     |      Default Value    |  Description |
+|:----------:|:-------------:|:------:       |
+| carbon.enable.index.server       |  false | Enable the use of index server for pruning        |
+| carbon.index.server.ip |    NA   |   Specify the IP/HOST on which the server would be started. Better to specify the private IP. | 
+| carbon.index.server.port | NA | The port on which the index server has to be started. |
+| carbon.disable.index.server.fallback | false | Whether to enable/disable fallback for index server. Should be used for testing purposes only |
+|carbon.index.server.max.worker.threads| 500 | Number of RPC handlers to open for accepting the requests from JDBC driver.|
+|carbon.max.executor.lru.cache.size|  NA | Used to specify the max size for executor LRU cache.|
+|carbon.index.server.max.jobname.length|NA|The max length of the job to show in the index server application UI. For bigger queries this may impact performance as the whole string would be sent from JDBCServer to IndexServer.|
+
+##### spark-defaults.conf(only for secure mode)
+
+| Name     |      Default Value    |  Description |
+|:----------:|:-------------:|:------:       |
+| spark.carbon.indexserver.principal |  NA | Used for authentication, whether a valid application is trying to connect to the server or not. Set in both IndexServer and JDBCServer     |
+| spark.carbon.indexserver.keytab |    NA   |   Specify the path to the keytab file through which authentication would happen. Set in both IndexServer and JDBCServer| 
+| spark.dynamicAllocation.enabled | true | Set to false, so that spark doesn’t kill the executor, If executors are killed, cache would be lost. Applicable only for Index Server. |
+| spark.yarn.principal | NA | Should be set to the same user used for JDBCServer.    |
+|spark.yarn.keytab| NA | Should be set to the same as JDBCServer.   |
+
+##### core-site.xml
+
+| Name     |      Default Value    |  Description |
+|:----------:|:-------------:|:------:       |
+| ipc.client.rpc-timeout.ms |  NA | Set the above property to some appropriate value based on your estimated query time. The best option is to set this to the same value as spark.network.timeout|
+ 
```

Comment:
spark-submit command to start the server can also be mentioned. Also below FAQs can be mentioned
1. Index Server is throwing Large response size exception.
2. Index server is throwing Kerberos principal not set exception
3. Unable to connect to index server
4. IndexServer is throwing FileNotFoundException for index files.

Reply:
done